crates/schema-api/src/lib.rs

```diff
@@ -86,6 +86,15 @@ pub mod deployment {
             #[cfg_attr(feature = "serde_schema", schemars(with = "String"))]
             address: Uri,
             protocol_type: ProtocolType,
+            #[cfg_attr(
+                feature = "serde",
+                serde(
+                    default,
+                    with = "serde_with::As::<Option<restate_serde_util::VersionSerde>>"
+                )
+            )]
+            #[cfg_attr(feature = "serde_schema", schemars(with = "Option<String>"))]
+            http_version: Option<http::Version>,
```

Comment:
It would have been really nice if we didn't leak the fact that this is a newly added field to the upper layers (e.g. if there is a possibility to resolve the http_version before creating the `DeploymentType` based on the `protocol_type`). That way the logic to read older values would be contained in a single place and spread to the places where `http_version` is used.

Reply:
I guess one way could be to have an intermediary type `OptionalDeploymentType` which has the optional fields and into which one serializes the `DeploymentType`. When deserializing `DeploymentType` one could read the `OptionalDeploymentType` first and then transform it into the `DeploymentType` with the non-optional fields.